flytekit/annotated/type_engine.py

```diff
@@ -391,31 +404,57 @@ def to_literal(
         python_type: Type[flyte_typing.FlyteFilePath],
         expected: LiteralType,
     ) -> Literal:
-        remote_path = ""
+        remote_path = ctx.file_access.get_random_remote_path()
+
         if isinstance(python_val, flyte_typing.FlyteFilePath):
-            remote_path = python_val.remote_path if python_val.remote_path else ""
+            if python_val.remote_path is False:
+                # If the user specified the remote_path to be False, that means no matter what, do not upload
+                remote_path = None
+            else:
+                # Otherwise, if not an "" use the user-specified remote path instead of the random one
+                remote_path = python_val.remote_path or remote_path
+            source_path = python_val.path
         else:
             if not (isinstance(python_val, os.PathLike) or isinstance(python_val, str)):
                 raise AssertionError(f"Expected FlyteFilePath or os.PathLike object, received {type(python_val)}")
+            source_path = python_val
 
-        if remote_path == "":
-            remote_path = ctx.file_access.get_random_remote_path()
+        # For remote values, say https://raw.github.com/demo_data.csv, we will not upload to Flyte's store (S3/GCS)
+        # and just return a literal with a uri equal to the path given
+        if ctx.file_access.is_remote(source_path):
+            # TODO: Add copying functionality so that FlyteFile(path="s3://a", remote_path="s3://b") will copy.
+            meta = BlobMetadata(type=self._blob_type(format=FlyteFilePathTransformer.get_format(python_type)))
+            return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=source_path)))
 
-        ctx.file_access.put_data(f"{python_val}", remote_path, is_multipart=False)
-        meta = BlobMetadata(type=self._blob_type(format=FlyteFilePathTransformer.get_format(python_type)))
-        return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=remote_path)))
+        # For local paths, we will upload to the Flyte store (note that for local execution, the remote store is just
+        # a subfolder), unless remote_path=False was given
+        else:
+            if remote_path is not None:
+                ctx.file_access.put_data(source_path, remote_path, is_multipart=False)
+            meta = BlobMetadata(type=self._blob_type(format=FlyteFilePathTransformer.get_format(python_type)))
+            return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=remote_path)))
 
     def to_python_value(
         self, ctx: FlyteContext, lv: Literal, expected_python_type: Type[flyte_typing.FlyteFilePath]
     ) -> flyte_typing.FlyteFilePath:
+
+        uri = lv.scalar.blob.uri
```

Comment:
this should never happen right?

Reply:
looks weird to me

Reply:
hmm?  is this comment on the right line?  I'm just making a local variable cuz there's too many dots.